### PR TITLE
Fix tokenizer kwarg in unifiedTransformer

### LINKED ISSUE #202: Cannot specify custom tokenizer with UnifiedTransformer
```py
!pip install -q git+https://github.com/ndif-team/nnsight@0.3 transformer_lens
from nnsight.models.UnifiedTransformer import UnifiedTransformer
from transformers import AutoTokenizer
tokenizer = AutoTokenizer.from_pretrained("gpt2")
model = UnifiedTransformer("gpt2", tokenizer=tokenizer, processing=False)


---> 43         super().__init__(hooked_model, tokenizer=self.tokenizer, *args, **kwargs)
     44 
     45     def _prepare_inputs(

TypeError: nnsight.models.LanguageModel.LanguageModel.__init__() got multiple values for keyword argument 'tokenizer'
```

Solution (PR incoming): check if tokenizer is in kwargs.

### PR DESCRIPTION
**Pull request by Butanium**
fixes #202

### CODE REVIEW

File: src/nnsight/models/UnifiedTransformer.py [hunk 1]
```diff
@@ -28,13 +28,13 @@ class UnifiedTransformer(LanguageModel):
...
+        self, model: str, *args, processing: bool = True, tokenizer: Optional[PreTrainedTokenizer] = None, **kwargs
     ) -> None:
         if processing:
-            hooked_model = HookedTransformer.from_pretrained(model, *args, **kwargs)
+            hooked_model = HookedTransformer.from_pretrained(model, *args, tokenizer=tokenizer, **kwargs)
         else:
             hooked_model = HookedTransformer.from_pretrained_no_processing(
-                model, *args, **kwargs
+                model, *args, tokenizer=tokenizer, **kwargs
             )
```
Comment (Butanium) [inline src/nnsight/models/UnifiedTransformer.py:39]: we could also 
```py
kwargs.pop("tokenizer", None)
```
here if you prefer

File: src/nnsight/models/UnifiedTransformer.py [hunk 2]
```diff
@@ -28,13 +28,13 @@ class UnifiedTransformer(LanguageModel):
...
 
         self.tokenizer = hooked_model.tokenizer
```
Comment (Butanium) [inline src/nnsight/models/UnifiedTransformer.py:40]: also why do you do that when it'll be done again in the LanguageModel init?

**Comment by Butanium** [comment]
Might be a good time to also fix https://github.com/ndif-team/nnsight/issues/138 by enforcing `padding_side="left"` for unified_transformer too ?

File: src/nnsight/models/UnifiedTransformer.py (hunk 2 above)
Comment (JadenFiotto-Kaufman) [reply src/nnsight/models/UnifiedTransformer.py:40]: Sorry what is the context behind this?

File: src/nnsight/models/UnifiedTransformer.py [hunk 3]
```diff
@@ -28,13 +28,13 @@ class UnifiedTransformer(LanguageModel):
     """
 
     def __init__(
-        self, model: str, *args, processing: bool = True, **kwargs
+        self, model: str, *args, processing: bool = True, tokenizer: Optional[PreTrainedTokenizer] = None, **kwargs
```
Comment (JadenFiotto-Kaufman) [inline src/nnsight/models/UnifiedTransformer.py:31]: You need to import Optional. After that I'll merge it in!

Comment (Butanium) [reply src/nnsight/models/UnifiedTransformer.py:31]: done!

File: src/nnsight/models/UnifiedTransformer.py (hunk 2 above)
Comment (Butanium) [reply src/nnsight/models/UnifiedTransformer.py:40]: in unifiedtransformer init, you set self.tokenizer and then call the super init which will run https://github.com/ndif-team/nnsight/blob/f64d8a963d72acd58f2eaae6b1a2fcbf4d29cb63/src/nnsight/models/LanguageModel.py#L141
which will set self.tokenizer again, right?

**Comment by JadenFiotto-Kaufman** [comment]
@Butanium Should we just get rid of enforcing padding side all together?

**Comment by Butanium** [comment]
@JadenFiotto-Kaufman I think we can probably do it but we'd need to ensure that the token indexing actually work in both cases

**Comment by Butanium** [comment]
@JadenFiotto-Kaufman could we merge this as is ? We can still take a decision regarding padding side later but it'd be nice to have this fixed asap

**Comment by JadenFiotto-Kaufman** [comment]
@Butanium 👍

**Pull request merged**